`hathora build` does not generate `dist` folder in custom react client

According to the [build documentation](https://docs.hathora.dev/#/deploy?id=build) `hathora build` should create a `dist` folder in _all_ client subdirectories, but in my project with a custom react client it does not.

`hathora build` output:
```
hathora build
Installing dependencies in /Users/martijnwelker/web/wizard/api

up to date, audited 7 packages in 456ms

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities

Installing dependencies in /Users/martijnwelker/web/wizard/client/.DS_Store
npm or yarn not found.
Installing dependencies in /Users/martijnwelker/web/wizard/client/.hathora

up to date, audited 27 packages in 545ms

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities

Installing dependencies in /Users/martijnwelker/web/wizard/client/prototype-ui

up to date, audited 131 packages in 536ms

22 packages are looking for funding
  run `npm fund` for details

1 moderate severity vulnerability

To address all issues, run:
  npm audit fix --force

Run `npm audit` for details.

Installing dependencies in /Users/martijnwelker/web/wizard/client/web
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: mini-create-react-context@0.4.1
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   react@"^18.2.0" from the root project
npm WARN   4 more (@testing-library/react, react-dom, react-router, react-router-dom)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"^0.14.0 || ^15.0.0 || ^16.0.0 || ^17.0.0" from mini-create-react-context@0.4.1
npm WARN node_modules/react-router/node_modules/mini-create-react-context
npm WARN   mini-create-react-context@"^0.4.0" from react-router@5.3.3
npm WARN   node_modules/react-router
npm WARN
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^0.14.0 || ^15.0.0 || ^16.0.0 || ^17.0.0" from mini-create-react-context@0.4.1
npm WARN   node_modules/react-router/node_modules/mini-create-react-context
npm WARN     mini-create-react-context@"^0.4.0" from react-router@5.3.3
npm WARN     node_modules/react-router

up to date, audited 1440 packages in 1s

205 packages are looking for funding
  run `npm fund` for details

5 high severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

Installing dependencies in /Users/martijnwelker/web/wizard/server

up to date, audited 2 packages in 1s

found 0 vulnerabilities

Installing dependencies in /Users/martijnwelker/web/wizard/server/.hathora

up to date, audited 14 packages in 456ms

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities


> build
> vite build

vite v2.9.9 building for production...
✓ 697 modules transformed.
dist/index.html                  6.48 KiB
dist/assets/index.97f32ad3.css   30.53 KiB / gzip: 6.42 KiB
dist/assets/index.39cabb85.js    293.49 KiB / gzip: 86.09 KiB
```

I've attempted to manually run `npm run build` in `client/web/` but that errors with:
```
$ npm run build --verbose
npm verb cli /Users/martijnwelker/.nvm/versions/node/v16.16.0/bin/node /Users/martijnwelker/.nvm/versions/node/v16.16.0/bin/npm
npm info using npm@8.14.0
npm info using node@v16.16.0
npm timing npm:load:whichnode Completed in 1ms
npm timing config:load:defaults Completed in 0ms
npm timing config:load:file:/Users/martijnwelker/.nvm/versions/node/v16.16.0/lib/node_modules/npm/npmrc Completed in 1ms
npm timing config:load:builtin Completed in 1ms
npm timing config:load:cli Completed in 1ms
npm timing config:load:env Completed in 0ms
npm timing config:load:file:/Users/martijnwelker/web/wizard/client/web/.npmrc Completed in 0ms
npm timing config:load:project Completed in 4ms
npm timing config:load:file:/Users/martijnwelker/.npmrc Completed in 1ms
npm timing config:load:user Completed in 1ms
npm timing config:load:file:/Users/martijnwelker/.nvm/versions/node/v16.16.0/etc/npmrc Completed in 0ms
npm timing config:load:global Completed in 0ms
npm timing config:load:validate Completed in 0ms
npm timing config:load:credentials Completed in 0ms
npm timing config:load:setEnvs Completed in 1ms
npm timing config:load Completed in 8ms
npm timing npm:load:configload Completed in 8ms
npm timing npm:load:mkdirpcache Completed in 0ms
npm timing npm:load:mkdirplogs Completed in 0ms
npm verb title npm run build
npm verb argv "run" "build" "--loglevel" "verbose"
npm timing npm:load:setTitle Completed in 6ms
npm timing config:load:flatten Completed in 1ms
npm timing npm:load:display Completed in 3ms
npm verb logfile logs-max:10 dir:/Users/martijnwelker/.npm/_logs
npm verb logfile /Users/martijnwelker/.npm/_logs/2022-09-01T13_07_44_327Z-debug-0.log
npm timing npm:load:logFile Completed in 2ms
npm timing npm:load:timers Completed in 0ms
npm timing npm:load:configScope Completed in 0ms
npm timing npm:load Completed in 20ms

> web@0.1.0 build
> node scripts/build.js

Creating an optimized production build...Completed in 20ms
Failed to compile.⠂) ⠴ : timing npm:load Completed in 20ms

Attempted import error: 'Writer' is not exported from 'bin-serde' (imported as '_Writer').


npm timing command:run Completed in 4792ms
npm verb exit 1
npm timing npm Completed in 4815ms
npm verb code 1
```

which comes from

```
import { Writer as _Writer, Reader as _Reader } from "bin-serde";
```

Extra info:
- `hathora --version`: 0.9.16
- `node -v`: v16.16.0
- `npm -v`: 8.14.0

If required I can make my repo public 😄 Any help would be appreciated

--- a/docs/deploy.md
+++ b/docs/deploy.md
@@ -12,8 +12,10 @@ The directory structure will look something like this:
 . # project root
 ├─ client
 │  ├─ prototype-ui
+│  │  ├─ package.json
 │  │  └─ dist
 │  └─ web
+│     ├─ package.json
 │     └─ dist
 └─ server
    └─ dist

--- a/src/commands/build.ts
+++ b/src/commands/build.ts
@@ -30,12 +30,15 @@ const cmd: CommandModule = {
   },
 };
 
-function build(only: "server" | "client" | undefined) {
+async function build(only: "server" | "client" | undefined) {
   const { clientDir, serverDir } = getDirs();
   if (only === "client" || only === undefined) {
     for (const dir of readdirSync(clientDir)) {
-      if (existsSync(join(clientDir, dir, "index.html"))) {
-        execSync("npm run build", { cwd: join(clientDir, dir), stdio: "inherit" });
+      if (existsSync(join(clientDir, dir, "package.json"))) {
+        const pkg = await import(join(clientDir, dir, "package.json"));
+        if (pkg.scripts?.start !== undefined) {
+          execSync("npm run build", { cwd: join(clientDir, dir), stdio: "inherit" });
+        }
       }
     }
   }

